src/openms/source/ANALYSIS/QUANTITATION/IsobaricQuantitationMethod.cpp

```diff
@@ -40,89 +40,57 @@
 namespace OpenMS
 {
   IsobaricQuantitationMethod::~IsobaricQuantitationMethod()
-  {
-  }
+  = default;
 
   IsobaricQuantitationMethod::IsobaricQuantitationMethod() :
     DefaultParamHandler("IsobaricQuantitationMethod")
   {
   }
 
-  Matrix<double> IsobaricQuantitationMethod::stringListToIsotopCorrectionMatrix_(const StringList& stringlist) const
+  Matrix<double> IsobaricQuantitationMethod::stringListToIsotopeCorrectionMatrix_(const StringList& stringlist) const
   {
     // check the string list
     if (stringlist.size() != getNumberOfChannels())
     {
       throw Exception::InvalidParameter(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, String("IsobaricQuantitationMethod: Invalid string representation of the isotope correction matrix. Expected ") + getNumberOfChannels() + " entries but got " + stringlist.size() + ".");
     }
 
-    // create matrix to fill from stringlist
-    Matrix<double> isotope_correction_matrix(getNumberOfChannels(), 4);
+    // compute frequency matrix based on the deviation matrix
+    Matrix<double> channel_frequency(getNumberOfChannels(), getNumberOfChannels(), 0.0);
 
     // channel index
-    Size channel_index = 0;
+    Size contributing_channel = 0;
 
     // fill row-wise
-    for (StringList::const_iterator it = stringlist.begin(); it != stringlist.end(); ++it)
+    for (const auto& l : stringlist)
     {
       StringList corrections;
-      it->split('/', corrections);
-      if (corrections.size() != 4)
-      {
-        throw Exception::InvalidParameter(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "IsobaricQuantitationMethod: Invalid entry in string representation of the isotope correction matrx. Expected four correction values separated by '/', got: '" + *it + "'");
-      }
+      l.split('/', corrections);
```

Comment:
Add a check, just to make sure that the user has entered 4 or 8 numbers per line (and not 9 or so...).

Warning: blindly coded :)

```suggestion
      l.split('/', corrections);
      auto number_of_columns = getChannelInformation()[contributing_channel].affected_channels.size();
      if (corrections.size() != number_of_columns )
      {
        throw Exception::InvalidValue(__FILE__, __LINE__, OPENMS_PRETTYFUNCTION, String("Entry #") + contributing_channel + " must contain " + number_of_columns + " values, but has " + corrections.size() + "!");
      }
```

Reply:
Could you please add a test, that this actually works? Thanks :)

Reply:
done. almost correct.